Add new logger and migrate multiple commands to allow json

mgdelacroix opened the pull request.

mattermod commented:
This issue has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @hanzei

hanzei commented:
@grundleborg Gentle reminder to review this

grundleborg commented on commands/channel.go:
```diff
@@ -167,10 +169,13 @@ func createChannelCmdF(command *cobra.Command, args []string) error {
 		return response.Error
 	}
 
+	printer.PrintT("New team {{.Name}} successfully created", newChannel)
```
```suggestion
	printer.PrintT("New channel {{.Name}} successfully created", newChannel)
```

grundleborg commented on commands/channel.go:
```diff
@@ -413,20 +418,21 @@ func searchChannelCmdF(command *cobra.Command, args []string) error {
 		channel, response = c.GetChannelByName(args[0], team.Id, "")
 		if response.Error != nil || channel == nil {
-			CommandPrettyPrintln(fmt.Sprintf("Channel %s is not found in team %s", args[0], teamArg))
+			printer.Print(fmt.Sprintf("Channel %s is not found in team %s", args[0], teamArg))
```
Should this use templates too?

grundleborg commented on printer/printer.go:
```diff
@@ -0,0 +1,59 @@
+func SetFormat(t string) {
+	printer.Format = t
+}
+
```
Would love it if you could add some godocs to the functions in this file. Reason I asked is because I didn't understand what `SetSingle()` meant from the function name alone.

mgdelacroix commented:
> Sorry for the incredibly slow review! Just some minor fixes/requests.

No worries!! I've addressed the comments, PTAL :)

> Also, one unrelated thought, should we use mmctl as an opportunity to standardise on "archive" rather than "delete" for soft-deleting things? We could also alias "delete" to "archive" in the commands for convenience, but this might be a chance to update all the language around that? If so, could be (potentially even help wanted) part of the mmctl epic.

Definitely, I've created [this ticket](https://mattermost.atlassian.net/browse/MM-19200) to track it.
